charms/istio-pilot/src/istio_gateway_name_provider.py

```diff
@@ -0,0 +1,25 @@
+import logging
+from ops.framework import Object, EventBase, EventSource, ObjectEvents
+from ops.charm import CharmEvents
+from lightkube.core.client import Client
+from lightkube.generic_resource import create_namespaced_resource
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_RELATION_NAME = "gateway"
+
+
+class GatewayProvider(Object):
+    def __init__(self, charm, relation_name=DEFAULT_RELATION_NAME):
+        super().__init__(charm, relation_name)
+
+    def on_gateway_relation_changed(self, charm, gateway_name, gateway_namespace):
```

Comment:
I think this name might be from when this actually handled the relation changed event.  Now it feels more like a `send_data` or something like that.